iceoryx_utils/include/iceoryx_utils/internal/concurrent/loffli.hpp

```diff
@@ -29,11 +29,14 @@ namespace concurrent
 {
 class LoFFLi
 {
+  public:
+    using Index_t = uint32_t;
+
   private:
     /// @todo std::atomic_is_lock_free check
     struct alignas(8) Node
     {
-        uint32_t indexToNextFreeIndex;
+        Index_t indexToNextFreeIndex;
```

Comment:
Could you please add either some runtime `cxx::Expects` that ensures that `LoFFLi` is lock free. Or a `static_assert` that `sizeof(Node) <= 8`.
When the `Index_t` is a type alias maybe some future smarty-pants (like me) thinks: "uuuh uint32_t is too small for me let's make it uint64_t" never realizes that LoFFLi is not lock free anymore.

Oh look there is already a "@todo" in the code for that :D

Reply:
If this is not possible here we could also add a test that verifies that LoFFLi is lock-free. For that, we would require only access to the `Node` struct but this can be achieved by replacing `private` with `protected` and then inherit from LoFFLi in a test mock.